android/src/main/AndroidManifest.xml

```diff
@@ -1,10 +1,10 @@
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
   package="com.alternadom.wifiiot">
-    <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE"/>
-    <uses-permission android:name="android.permission.ACCESS_WIFI_STATE"/>
-    <uses-permission android:name="android.permission.CHANGE_NETWORK_STATE"/>
-    <uses-permission android:name="android.permission.CHANGE_WIFI_STATE" />
-    <uses-permission android:name="android.permission.WRITE_SETTINGS" />
+    <uses-feature android:name="android.hardware.wifi" />
     <uses-permission android:name="android.permission.INTERNET" />
+    <uses-permission android:name="android.permission.ACCESS_WIFI_STATE"/>
+    <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE" />
+    <uses-permission android:name="android.permission.CHANGE_NETWORK_STATE" />
     <uses-permission android:name="android.permission.ACCESS_FINE_LOCATION" />
+    <uses-permission android:name="android.permission.ACCESS_COARSE_LOCATION" />
```

Comment:
These is a breaking change, user needs to now explicitly add `WRITE_SETTINGS`, `CHANGE_WIFI_STATE` and `CHANGE_NETWORK_STATE` if they use certain functions.

For reviewers, should we have this breaking change? Or add all the permission by default. User can remove the permission if they want in there manifest with something like following:

```xml
<uses-permission
        android:name="android.permission.WRITE_SETTINGS"
        tools:node="remove"/>
```

Reply:
If all permissions are added, the user must delete the permission from the library. To prevent modifying the library's code, permissions for unique functions like `WRITE_SETTINGS`, `CHANGE_WIFI_STATE`, and `CHANGE_NETWORK_STATE` can be inserted manually in the user's project. This way, the user will concentrate on their app instead of thinking about deleting the unused permissions. That is my viewpoint. You are free to disagree with me.

Reply:
@mavyfaby, I agree with you (I actually requested this change!), but the issue here is that this is a breaking change, current user need to know this.

We might have to bump to `1.0.0` in this case, if we go ahead with removing this permissions from plugin manifest.

Reply:
so just taking a vote here!

Reply:
Yes, we must specify it in the README on the introduction page so that users are aware of the latest changes.

Reply:
I will add the note, once @fedepell and @Niek gives there view on it.

Reply:
will release this time with all permission[e1db081e6b92216e8c168e31cd23db824a087d3b], since this is breaking change, will release this when we do `v1.0.0` - as of now added `TODO`s.

Reply:
Resolving this, feel free to open this discussion.